src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -538,3 +542,58 @@ export const updateFormSettings = (
     return okAsync(updatedForm.getSettings())
   })
 }
+
+/**
+ * Deletes form logic.
+ * @param form The original form to delete logic in
+ * @param logicId the logicId to delete
+ * @returns ok(true) on success
+ * @returns err(database errors) if db error is thrown during logic delete
+ * @returns err(LogicNotFoundError) if logicId does not exist on form
+ */
+export const deleteFormLogic = (
+  form: IPopulatedForm,
+  logicId: string,
+): Result<never, LogicNotFoundError> | ResultAsync<true, DatabaseError> => {
+  // First check if specified logic exists
+  if (!form.form_logics.some((logic) => logic.id === logicId)) {
+    logger.error({
+      message: 'Error occurred when deleting form logic',
+      meta: {
+        action: 'deleteFormLogic',
+        formId: form._id,
+        logicId,
+      },
+    })
+    return err(new LogicNotFoundError())
+  }
+
+  // Remove specified logic and then update form logic
+  const ModelToUse = getFormModel(mongoose)
```

Comment:
nit: call this `FormModel`?

nonnit: should we instantiate this outside of this function so this retrieval of model only gets called a single time?

Reply:
edited